From 9f4753af46e77aa355a11674d5c3171e703ca14a Mon Sep 17 00:00:00 2001
From: Artem Lapkin <art@khadas.com>
Date: Wed, 26 May 2021 12:54:07 +0800
Subject: [PATCH] spl extra verbose uboot source

Signed-off-by: Artem Lapkin <art@khadas.com>
---
 common/spl/spl.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/common/spl/spl.c b/common/spl/spl.c
index a0a608fd..ec2b42d8 100644
--- a/common/spl/spl.c
+++ b/common/spl/spl.c
@@ -597,6 +597,16 @@ static int boot_from_devices(struct spl_image_info *spl_image,
 			     u32 spl_boot_list[], int count)
 {
 	int i;
+	printf("spl_boot: ");
+	for (i = 0; i < count && spl_boot_list[i] != BOOT_DEVICE_NONE; i++) {
+		struct spl_image_loader *loader;
+		loader = spl_ll_find_loader(spl_boot_list[i]);
+		if (loader)
+			printf("%s > ", loader->name);
+		else
+			puts("?? > ");
+	}
+	printf("\n");
 
 	for (i = 0; i < count && spl_boot_list[i] != BOOT_DEVICE_NONE; i++) {
 		struct spl_image_loader *loader;
-- 
2.25.1

