From aa72295faeb4f2d2dc8eb56422f3b20c1bec08dc Mon Sep 17 00:00:00 2001
From: Artem Lapkin <art@khadas.com>
Date: Tue, 23 Feb 2021 13:41:55 +0800
Subject: [PATCH] tftp set env tftpip from last source

---
 net/tftp.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/net/tftp.c b/net/tftp.c
index 6fdb1a82..896c9b0a 100644
--- a/net/tftp.c
+++ b/net/tftp.c
@@ -798,6 +798,10 @@ void tftp_start(enum proto_t protocol)
 #endif
 	       &tftp_remote_ip, &net_ip);
 
+	char tmp[22];
+	ip_to_string(tftp_remote_ip, tmp);
+	env_set("tftpip", tmp);
+
 	/* Check if we need to send across this subnet */
 	if (net_gateway.s_addr && net_netmask.s_addr) {
 		struct in_addr our_net;
-- 
2.25.1

