From 2dbbb7a50c7efa0cae5c1046fffea88b9efdce57 Mon Sep 17 00:00:00 2001
From: Artem Lapkin <art@khadas.com>
Date: Mon, 26 Apr 2021 16:17:43 +0800
Subject: [PATCH] ignored booted from other source for mmc

Signed-off-by: Artem Lapkin <art@khadas.com>
---
 env/mmc.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/env/mmc.c b/env/mmc.c
index ee376c3e..0cc12598 100644
--- a/env/mmc.c
+++ b/env/mmc.c
@@ -306,6 +306,11 @@ static int env_mmc_load(void)
 	int dev = mmc_get_env_dev();
 	const char *errmsg = NULL;
 
+	if (dev < 0 ) {
+	    printf("ignored booted from other source!\n");
+	    return 1;
+	}
+
 	ALLOC_CACHE_ALIGN_BUFFER(env_t, tmp_env1, 1);
 	ALLOC_CACHE_ALIGN_BUFFER(env_t, tmp_env2, 1);
 
@@ -352,6 +357,11 @@ static int env_mmc_load(void)
 	const char *errmsg;
 	env_t *ep = NULL;
 
+	if (dev < 0 ) {
+	    printf("ignored booted from other source!\n");
+	    return 1;
+	}
+
 	mmc = find_mmc_device(dev);
 
 	errmsg = init_mmc_for_env(mmc);
-- 
2.25.1

