From 8a957ee249c27225a6216fa5d55a6a02cd201955 Mon Sep 17 00:00:00 2001
From: Lapkin Artem <art@khadas.com>
Date: Fri, 4 Dec 2020 11:05:59 +0800
Subject: [PATCH 23/43] khadas_logo

diff --git a/include/configs/meson64.h b/include/configs/meson64.h
index 6c107870..79338426 100644
--- a/include/configs/meson64.h
+++ b/include/configs/meson64.h
@@ -69,6 +69,24 @@
 	func(DHCP, dhcp, na)
 #endif
 
+#define CONFIG_SYS_VIDEO_LOGO_MAX_SIZE 8192000
+#define CONFIG_VIDEO_BMP_GZIP 1
+
+#define CONSOLE_FONT_COLOR 14
+
+#define PREBOOT_LOAD_LOGO \
+	"ll=0; test $boot_source = spi && sf probe && sf read $loadaddr 0x170000 0x10000 && ll=1; " \
+	"test $ll = 0 && ll=1 && " \
+	"load mmc 1 $loadaddr splash.bmp || " \
+	"load mmc 2 $loadaddr splash.bmp || " \
+	"load mmc 1:2 $loadaddr /usr/share/fenix/logo/logo.bmp || " \
+	"load mmc 2:2 $loadaddr /usr/share/fenix/logo/logo.bmp || " \
+	"ll=0; " \
+	"test $ll = 1 && bmp display $loadaddr m m || ll=0; " \
+	"test $ll = 0 && fdt addr $fdtcontroladdr && fdt get addr logoaddr /logo data && bmp display $logoaddr m m && echo [i] display embed logo; "
+
+#define PREBOOT_CMD "run load_logo; usb start; kbi init; sleep 1;"
+
 #define CONFIG_HOSTNAME CONFIG_DEFAULT_DEVICE_TREE
 #define CONFIG_BOOTP_SEND_HOSTNAME 1
 
@@ -76,10 +94,13 @@
 
 #ifndef CONFIG_EXTRA_ENV_SETTINGS
 #define CONFIG_EXTRA_ENV_SETTINGS \
+	"load_logo=" PREBOOT_LOAD_LOGO "\0" \
+	"preboot=" PREBOOT_CMD "\0" \
 	"stdin=" STDIN_CFG "\0" \
 	"stdout=" STDOUT_CFG "\0" \
 	"stderr=" STDOUT_CFG "\0" \
 	"fdt_addr_r=0x08008000\0" \
+	"loadaddr=0x01000000\0" \
 	"scriptaddr=0x08000000\0" \
 	"kernel_addr_r=0x08080000\0" \
 	"pxefile_addr_r=0x01080000\0" \
-- 
2.17.1

