From 682bb0029aa9e7b2345ac8e14ddd8ecbac9bf4cd Mon Sep 17 00:00:00 2001
From: Artem Lapkin <art@khadas.com>
Date: Mon, 24 May 2021 13:45:22 +0800
Subject: [PATCH] meson64: add kernel_comp configs

Signed-off-by: Artem Lapkin <art@khadas.com>
---
 include/configs/meson64.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/configs/meson64.h b/include/configs/meson64.h
index e3d25493..552a08c2 100644
--- a/include/configs/meson64.h
+++ b/include/configs/meson64.h
@@ -113,6 +113,8 @@
 	"stdin=" STDIN_CFG "\0" \
 	"stdout=" STDOUT_CFG "\0" \
 	"stderr=" STDOUT_CFG "\0" \
+	"kernel_comp_addr_r=0x0d080000\0" \
+	"kernel_comp_size=0x2000000\0" \
 	"fdt_addr_r=0x08008000\0" \
 	"loadaddr=0x01000000\0" \
 	"scriptaddr=0x08000000\0" \
-- 
2.25.1

