From b3458f39789cc741dd7514f59038eb8abe31a4e5 Mon Sep 17 00:00:00 2001
From: Artem Lapkin <art@khadas.com>
Date: Mon, 1 Mar 2021 13:13:42 +0800
Subject: [PATCH] no_usb env

Signed-off-by: Artem Lapkin <art@khadas.com>
---
 cmd/usb.c             | 5 +++++
 drivers/core/uclass.c | 8 ++++++++
 2 files changed, 13 insertions(+)

diff --git a/cmd/usb.c b/cmd/usb.c
index b9ec29a8..a3e1f8f4 100644
--- a/cmd/usb.c
+++ b/cmd/usb.c
@@ -632,6 +632,11 @@ static int do_usb(struct cmd_tbl *cmdtp, int flag, int argc, char *const argv[])
 	int i;
 	extern char usb_started;
 
+	if (env_get("no_usb")){
+	    printf("!no_usb - was defined\n");
+	    return 0;
+	}
+
 	if (argc < 2)
 		return CMD_RET_USAGE;
 
-- 
2.25.1

