From 091cc0c8e8c774310cc1be7298c5f5b49ad6bda4 Mon Sep 17 00:00:00 2001
From: Artem Lapkin <art@khadas.com>
Date: Tue, 8 Jun 2021 18:15:42 +0800
Subject: [PATCH 1/2] mach-rockchip spl add boot_source

Signed-off-by: Artem Lapkin <art@khadas.com>
---
 arch/arm/mach-rockchip/spl.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/mach-rockchip/spl.c b/arch/arm/mach-rockchip/spl.c
index 02c40fb3..5ba2026b 100644
--- a/arch/arm/mach-rockchip/spl.c
+++ b/arch/arm/mach-rockchip/spl.c
@@ -35,6 +35,11 @@ const char *board_spl_was_booted_from(void)
 	u32  bootdevice_brom_id = readl(BROM_BOOTSOURCE_ID_ADDR);
 	const char *bootdevice_ofpath = NULL;
 
+	// save value+100 to safety mem area 0x01000000
+	// for next boot_source usage in uboot stage
+	writel(bootdevice_brom_id + 100, 0x01000000);
+	printf("*** BOOT_SOURCE_ID %u (2:emmc 3:spi 5:sd ...)***\n", bootdevice_brom_id );
+
 	if (bootdevice_brom_id < ARRAY_SIZE(boot_devices))
 		bootdevice_ofpath = boot_devices[bootdevice_brom_id];
 
-- 
2.25.1

