From e99a75c18bc11c59d800f36dfa69203e7012bf9f Mon Sep 17 00:00:00 2001
From: Artem Lapkin <art@khadas.com>
Date: Fri, 4 Jun 2021 12:52:40 +0800
Subject: [PATCH] fix pci meson_pcie_link_up fail bug

Signed-off-by: Artem Lapkin <art@khadas.com>
---
 drivers/pci/pcie_dw_meson.c | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/drivers/pci/pcie_dw_meson.c b/drivers/pci/pcie_dw_meson.c
index 0525ecbe..3d9a18eb 100644
--- a/drivers/pci/pcie_dw_meson.c
+++ b/drivers/pci/pcie_dw_meson.c
@@ -262,6 +262,8 @@ static int meson_pcie_init_port(struct udevice *dev)
 	int ret;
 	struct meson_pcie *priv = dev_get_priv(dev);
 
+	printf("%s\n", __func__);
+
 	ret = generic_phy_init(&priv->phy);
 	if (ret) {
 		dev_err(dev, "failed to init phy (ret=%d)\n", ret);
@@ -321,13 +323,14 @@ static int meson_pcie_init_port(struct udevice *dev)
 
 	ret = meson_pcie_link_up(priv, LINK_SPEED_GEN_2);
 	if (ret < 0)
-		goto err_link_up;
+		printf("meson_pci_link_up (ret=%d)\n", ret);
+//		goto err_link_up;
 
 	return 0;
-err_link_up:
-	clk_disable(&priv->clk_port);
-	clk_disable(&priv->clk_general);
-	clk_disable(&priv->clk_pclk);
+//err_link_up:
+//	clk_disable(&priv->clk_port);
+//	clk_disable(&priv->clk_general);
+//	clk_disable(&priv->clk_pclk);
 err_deassert_bulk:
 	reset_assert_bulk(&priv->rsts);
 err_power_off_phy:
-- 
2.25.1

