From f8f085a13f084e3835aeed5e5d7c1b95a92468a1 Mon Sep 17 00:00:00 2001
From: Artem Lapkin <art@khadas.com>
Date: Fri, 30 Jul 2021 13:36:32 +0800
Subject: [PATCH] efi: console: force 80x25 and replace all unicode chars to .

Signed-off-by: Artem Lapkin <art@khadas.com>
---
 lib/efi_loader/efi_console.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/lib/efi_loader/efi_console.c b/lib/efi_loader/efi_console.c
index 3b012e1a..70bc6fd6 100644
--- a/lib/efi_loader/efi_console.c
+++ b/lib/efi_loader/efi_console.c
@@ -163,7 +163,11 @@ static efi_status_t EFIAPI efi_cout_output_string(
 	}
 	pos = buf;
 	utf16_utf8_strcpy(&pos, string);
+	if ( strlen(buf) > 1 ) {
+	fputs(stdout, ".");
+	} else {
 	fputs(stdout, buf);
+	}
 	free(buf);
 
 	/*
@@ -349,6 +353,9 @@ static void query_console_size(void)
 	if (ret)
 		return;
 
+	rows=25;
+	cols=80;
+
 	/* Test if we can have Mode 1 */
 	if (cols >= 80 && rows >= 50) {
 		efi_cout_modes[1].present = 1;
-- 
2.25.1

